### ISSUE TITLE
Synonyms missing in synonym output

### ISSUE THREAD

**Issue by gaurav** [TranslatorSRI/Babel#128]
As of PR #113, there are some missing synonyms in the synonym output. For example, NCBIGene:125273804 has the following label and synonym information:

```shell
nru@babel:/code/babel/babel_downloads/NCBIGene$ grep "NCBIGene:125273804" labels 
NCBIGene:125273804	fkbp5
nru@babel:/code/babel/babel_downloads/NCBIGene$ grep "NCBIGene:125273804" synonyms
NCBIGene:125273804	-
NCBIGene:125273804	peptidyl-prolyl cis-trans isomerase FKBP5
NCBIGene:125273804	FKBP prolyl isomerase 5
NCBIGene:125273804	fkbp5
```

But none of the other synonyms make their way into the synonyms file.

```json
{"curie": "NCBIGene:125273804", "names": [], "types": ["Gene", "GeneOrGeneProduct", "GenomicEntity", "ChemicalEntityOrGeneOrGeneProduct", "PhysicalEssence", "OntologyClass", "BiologicalEntity", "ThingWithTaxon", "NamedThing", "Entity", "PhysicalEssenceOrOccurrent", "MacromolecularMachineMixin"], "preferred_name": "fkbp5"}
```

Note that the label is correctly included in the compendium file:

```
{"type": "biolink:Gene", "identifiers": [{"i": "NCBIGene:125273804", "l": "fkbp5"}]}
```

Unrelatedly: note that NCBIGene labels [NCBIGene:125273804](https://www.ncbi.nlm.nih.gov/gene/?term=125273804) as `fkbp5\tFKBP prolyl isomerase 5 [ Megalobrama amblycephala (Wuchang bream) ]` -- I wonder if we could/should generate the more detailed name somehow?

**Comment by gaurav** [comment]
This appears to be caused by the NCBIGene synonym file being in the format: `id\tsynonym` rather than `id\trelation\tsynonym` (which I've now fixed in 5587a5c0baf) -- but it's been that way for a while, so this probably isn't the cause of the lack of synonyms we're seeing in the current NameRes.

**Comment by gaurav** [comment]
This actually appears to have been caused by the SynonymFactory using `Text.get_curie()`, which -- unlike `Text.get_prefix()` -- converts the prefix to uppercase:  

https://github.com/TranslatorSRI/Babel/blob/41d62bed0c009c0a62e45b91b209e854cee0831a/src/util.py#L75-L85

This causes a problem in SynonymFactory, which attempts to read the labels from `babel_downloads/NCBIGENE/labels` instead of the correct `babel_downloads/NCBIGene/labels`. I'm not sure where we actually need to do case-insensitive-prefix-matching, but I think we should maybe rename this method to `get_prefix_uc()` or (my preference) turn it into lowercase and replace this with `get_prefix_lc()`.
